### ISSUE TITLE
ClientHello Message

### ISSUE THREAD

**Issue by SteveSelva** [facebook/proxygen#447]
I want to generate SSL Certificate on the fly. I found a way using the ClientHello message in the TLS handshake, which contains the Server Name in the SNI extension. Is there any way to get the Server Name at ClientHello message or any other way to generate Certificate on the fly?

**Comment by afrind** [comment]
There used to be a hook for this in wangle::SSLContextConfig, but it was removed in https://github.com/facebook/wangle/commit/398695af7214db7930fe0104367dfb4071669996

**Comment by afrind** [comment]
CC: @AjanthanAsogamoorthy

**Comment by SteveSelva** [comment]
I have found two functions in `folly::SSLContext`; `folly::SSLContext::setServerNameCallback()` & `folly::SSLContext::addClientHelloCallback()`, which purposes my functionality, but to use those I must use `folly::AsyncSSLSocket::sslAccept` to accept the connection instead of using `Fizz`.
But I don't know how to use it properly, or its with a bug, `folly::AsyncSSLSocket::sslAccept()`. And I have raised it as an issue in Facebook::folly github group.[https://github.com/facebook/folly/issues/2007](url). Help me fix this issue.
